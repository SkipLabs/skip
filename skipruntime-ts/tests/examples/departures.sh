#!/bin/bash
set -e
cd ../../examples

cleanup() {
    jobs -p | xargs kill
}

trap cleanup EXIT

if [ "$#" -lt 2 ]; then
    echo "Usage: $0 <stdout target> <stderr target> <platform>"
fi

if [ "$3" = "native" ]; then
    echo "Run departures with native platform"
    LD_LIBRARY_PATH=$(realpath ../../build/skipruntime) SKIP_PLATFORM="native" node dist/departures.js >/dev/null &
else
    echo "Run departures with wasm platform"
    node dist/departures.js >/dev/null &
fi

sleep 1 # give a moment for service to spin up
node dist/departures-client.js >"$1" 2>"$2"

