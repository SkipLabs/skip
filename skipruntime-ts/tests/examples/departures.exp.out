[[0,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"2565","year":"2016"}]],[1,[{"asylum":"JOR","asylum_name":"Jordan","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"17","year":"2016"}]],[2,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"479","year":"2016"}]],[3,[{"asylum":"JOR","asylum_name":"Jordan","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"10725","year":"2016"}]],[4,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"1869","year":"2017"}]],[5,[{"asylum":"JOR","asylum_name":"Jordan","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"259","year":"2017"}]],[6,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"201","year":"2017"}]],[7,[{"asylum":"JOR","asylum_name":"Jordan","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"1225","year":"2017"}]]]
Init [
  [ 0, [ [Object] ] ],
  [ 1, [ [Object] ] ],
  [ 2, [ [Object] ] ],
  [ 3, [ [Object] ] ],
  [ 4, [ [Object] ] ],
  [ 5, [ [Object] ] ],
  [ 6, [ [Object] ] ],
  [ 7, [ [Object] ] ]
]
Update [
  [ 0, [ [Object] ] ],
  [ 1, [ [Object] ] ],
  [ 2, [ [Object] ] ],
  [ 3, [ [Object] ] ]
]
[[0,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"1094","year":"2015"}]],[1,[{"asylum":"JOR","asylum_name":"Jordan","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"200","year":"2015"}]],[2,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"123","year":"2015"}]],[3,[{"asylum":"JOR","asylum_name":"Jordan","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"1433","year":"2015"}]]]
Update [
  [ 0, [ [Object] ] ],
  [ 1, [ [Object] ] ],
  [ 2, [ [Object] ] ],
  [ 3, [ [Object] ] ]
]
[[0,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"1094","year":"2015"}]],[1,[{"asylum":"JOR","asylum_name":"Jordan","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"200","year":"2015"}]],[2,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"123","year":"2015"}]],[3,[{"asylum":"JOR","asylum_name":"Jordan","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"1433","year":"2015"}]]]
Update [ [ 0, [ [Object] ] ], [ 1, [ [Object] ] ] ]
[[0,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"123","year":"2015"}]],[1,[{"asylum":"JOR","asylum_name":"Jordan","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"1433","year":"2015"}]]]
Update [
  [ 0, [ [Object] ] ],
  [ 1, [ [Object] ] ],
  [ 2, [ [Object] ] ],
  [ 3, [ [Object] ] ],
  [ 4, [ [Object] ] ],
  [ 5, [ [Object] ] ],
  [ 6, [ [Object] ] ],
  [ 7, [ [Object] ] ]
]
[[0,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"2565","year":"2016"}]],[1,[{"asylum":"JOR","asylum_name":"Jordan","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"17","year":"2016"}]],[2,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"479","year":"2016"}]],[3,[{"asylum":"JOR","asylum_name":"Jordan","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"10725","year":"2016"}]],[4,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"1869","year":"2017"}]],[5,[{"asylum":"JOR","asylum_name":"Jordan","destination":"NOR","destination_name":"Norway","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"259","year":"2017"}]],[6,[{"asylum":"LEB","asylum_name":"Lebanon","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"201","year":"2017"}]],[7,[{"asylum":"JOR","asylum_name":"Jordan","destination":"USA","destination_name":"United States of America","origin":"SYR","origin_name":"Syrian Arab Rep.","persons":"1225","year":"2017"}]]]
